forc-test/src/lib.rs

```diff
@@ -857,6 +862,42 @@ mod tests {
         build(build_options)
     }
 
+    /// Build the tests in the test contract located at
+    /// "CARGO_MANIFEST_DIR/TEST_DATA_FOLDER_NAME/TEST_CONTRACT_PACKAGE_NAME".
+    fn test_contract_built_tests() -> anyhow::Result<BuiltTests> {
+        let cargo_manifest_dir = env!("CARGO_MANIFEST_DIR");
+        let contract_package_dir = PathBuf::from(cargo_manifest_dir)
+            .join(TEST_DATA_FOLDER_NAME)
+            .join(TEST_CONTRACT_PACKAGE_NAME);
+        let contract_package_dir_string = contract_package_dir.to_string_lossy().to_string();
+        let build_options = Opts {
+            pkg: forc_pkg::PkgOpts {
+                path: Some(contract_package_dir_string),
+                ..Default::default()
+            },
+            ..Default::default()
+        };
+        build(build_options)
+    }
```

Comment:
nit: we could DRY this up by having one function `test_built_tests(package_name: &str)`